test/protocol-buffer-conversion-test.ts

```diff
@@ -220,4 +228,240 @@ describe('Protocol Buffer Conversion', function(): void {
+    currencyAmountProto.setIssuedCurrencyAmount(testInvalidIssuedCurrency)
+    const amountProto = new Amount()
+    amountProto.setValue(currencyAmountProto)
+
+    // destination
+    const accountAddressProto = new AccountAddress()
+    accountAddressProto.setAddress('r123')
+    const destinationProto = new Destination()
+    destinationProto.setValue(accountAddressProto)
+
+    const paymentProto = new Payment()
+    paymentProto.setAmount(amountProto)
+    paymentProto.setDestination(destinationProto)
+
+    // WHEN the protocol buffer is converted to a native TypeScript type THEN the result is undefined
+    assert.isUndefined(XRPPayment.from(paymentProto))
+  })
+
+  it('Convert Payment with invalid deliverMin field', function(): void {
+    // GIVEN a payment protocol buffer with an invalid deliverMin field
+    // amount
+    const currencyAmountProto = new CurrencyAmount()
+    currencyAmountProto.setIssuedCurrencyAmount(testIssuedCurrency)
+    const amountProto = new Amount()
+    amountProto.setValue(currencyAmountProto)
+
+    // destination
+    const accountAddressProto = new AccountAddress()
+    accountAddressProto.setAddress('r123')
+    const destinationProto = new Destination()
+    destinationProto.setValue(accountAddressProto)
+
+    // deliverMin (invalid)
+    const deliverMinCurrencyAmountProto = new CurrencyAmount()
+    deliverMinCurrencyAmountProto.setIssuedCurrencyAmount(
+      testInvalidIssuedCurrency,
+    )
+    const deliverMinProto = new DeliverMin()
+    deliverMinProto.setValue(deliverMinCurrencyAmountProto)
+
+    const paymentProto = new Payment()
+    paymentProto.setAmount(amountProto)
+    paymentProto.setDestination(destinationProto)
+    paymentProto.setDeliverMin(deliverMinProto)
+
+    // WHEN the protocol buffer is converted to a native TypeScript type THEN the result is undefined
+    assert.isUndefined(XRPPayment.from(paymentProto))
+  })
+
+  it('Convert Payment with invalid sendMax field', function(): void {
+    // GIVEN a payment protocol buffer with an invalid sendMax field
+    // amount
+    const currencyAmountProto = new CurrencyAmount()
+    currencyAmountProto.setIssuedCurrencyAmount(testIssuedCurrency)
+    const amountProto = new Amount()
+    amountProto.setValue(currencyAmountProto)
+
+    // destination
+    const accountAddressProto = new AccountAddress()
+    accountAddressProto.setAddress('r123')
+    const destinationProto = new Destination()
+    destinationProto.setValue(accountAddressProto)
+
+    // sendMax (invalid)
+    const sendMaxCurrencyAmountProto = new CurrencyAmount()
+    sendMaxCurrencyAmountProto.setIssuedCurrencyAmount(
+      testInvalidIssuedCurrency,
+    )
+    const sendMaxProto = new SendMax()
+    sendMaxProto.setValue(sendMaxCurrencyAmountProto)
+
+    const paymentProto = new Payment()
+    paymentProto.setAmount(amountProto)
+    paymentProto.setDestination(destinationProto)
+    paymentProto.setSendMax(sendMaxProto)
```

Comment:
This is a lot of boilerplate for the `GIVEN`s in these tests. Is it worth writing a helper method to take care of these? (Your call).

Reply:
I think the reason to wait on a helper method like this is because each test case is creating a slightly different Payment proto in terms of which fields are actually set and what their values are, so for the moment I think the overhead of that helper wouldn't be worth it.

Reply:
With too many more test cases though... it probably would because it is such a pain in the ass to set all of these things up without the option for chaining.